api/filters/platform_termination_filter.go

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package filters
+
+import (
+	"github.com/Peripli/service-manager/pkg/query"
+	"github.com/Peripli/service-manager/pkg/types"
+	"github.com/Peripli/service-manager/pkg/util"
+	"github.com/Peripli/service-manager/pkg/web"
+	"github.com/Peripli/service-manager/storage"
+	"net/http"
+)
+
+const (
+	PlatformTerminationFilterName = "PlatformTerminationFilter"
+)
+
+func NewPlatformTerminationFilter(repository storage.Repository) *platformTerminationFilter {
+	return &platformTerminationFilter{
+		repository: repository,
+	}
+}
+
+// platformTerminationFilter ensures that platform provided is considered as not active and only then deletion is possible.
+type platformTerminationFilter struct {
+	repository storage.Repository
+}
+
+func (*platformTerminationFilter) Name() string {
+	return PlatformTerminationFilterName
+}
+
+func (f *platformTerminationFilter) Run(req *web.Request, next web.Handler) (*web.Response, error) {
+	platformID := req.PathParams[web.PathParamResourceID]
+
+	switch req.Request.Method {
+	case http.MethodDelete:
+		if platformID != "" {
```

Comment:
do we support DELETE /v1/platforms?fieldQuery=<whatever>? if so, this filter will not be applied in that flow